pkg/bootstrap/platform/gcp.go

```diff
@@ -251,15 +252,19 @@ func waitForMetadataSuppliers(suppliers []metadataSupplier, md map[string]string
 
 // Converts a GCP zone into a region.
 func zoneToRegion(z string) (string, error) {
-	// Zones are in the form <region>-<zone_suffix>, so capture everything but the suffix.
-	re := regexp.MustCompile("(.*)-.*")
-	m := re.FindStringSubmatch(z)
+	m := zoneRE.FindStringSubmatch(z)
 	if len(m) != 2 {
 		return "", fmt.Errorf("unable to extract region from GCP zone: %s", z)
 	}
 	return m[1], nil
 }
 
+var (
+	// Zones are in the form <country>-<region>-<zone_suffix>, so capture everything but the suffix.
+	zoneRE = regexp.MustCompile(`^([^-]*-[^-]*)-[^-]*$`)
+	zone   atomic.String
```

Comment:
We probably don't want to use a global variable here

Reply:
for the `zone` I moved to the gcpEnv struct.
But, I kept `zoneRE` as global var since:
1.  it is shared by two functions
2. the errors from `regexp.MustCompile` can be detected when compiling

Reply:
Yep I am good on the regex, just not `zone`. thanks